rfcs/0085-nixos-release-stablization.md

```diff
@@ -0,0 +1,128 @@
+---
+feature: nixos-release-stablization
+start-date: 2021-01-17
+author: Jonathan Ringer (@jonringer)
+co-authors:
+shepherd-team: @ryantm, @garbas, @Mic92
+shepherd-leader: @ryantm
+related-issues: [NixOS release schedule](https://github.com/NixOS/rfcs/pull/80)
+---
+
+# Summary
+[summary]: #summary
+
+To bring more certainty to the release cycle, add short periods where
+breaking changes are partially restricted. A list of Release Critical
+Packages is defined. Also, move most release stablization work from
+the `release` branch to the `master` branch to reduce backports.
+
+# Motivation
+[motivation]: #motivation
+
+We want to release on time without herculian efforts. Limited
+restrictions to merging breaking changes should avoid the common
+practice of forcing many, largely-untested, commits into master before
+branch-off.
+
+This happened for the 20.09 release in which ~640 staging
+commits were pushed through in the days leading up to the branch off
+([#95492](https://github.com/NixOS/nixpkgs/pull/95492),
+[#96280](https://github.com/NixOS/nixpkgs/pull/96280),
+[#96437](https://github.com/NixOS/nixpkgs/pull/96437),
+[#97146](https://github.com/NixOS/nixpkgs/pull/97146)). The last PR included a
+release blocker fix (introduced in [one of the mentioned
+PRs](https://github.com/NixOS/nixpkgs/pull/96437)) which delayed the
+branch-off from 4 Sept 2020 to 7 Sept 2020, and ZHF had to be delayed
+another day to get a semi-accurate hydra build status. The systemd
+bump was also largely responsible delaying the release until after the
+scheduled release date where basic login features in plasma [weren't
+restored until Oct 5th, a week after the planned release
+date](https://github.com/NixOS/nixpkgs/pull/99629) and a backlog of
+other plasma related issues [weren't resolved until Oct 19th, almost a
+month after the planned release
+date](https://github.com/NixOS/nixpkgs/pull/101078).  This is not to
+discredit any individual, but demonstrate that certain packages have
+many dimensions of compatibility which need a longer time to stabilize
+on unstable before being included in a release.
+
+
+# Definitions
+[definitions]: #definition
+- Breaking change
+  - A change that is likely to break downstream.
+  - For projects following Semantic Versioning, disallow major version bumps.
```

Comment:
The reference to Semantic Versioning was to further explain what a "breaking change" was, since semver already captures this concept stable releases in the form of a major version number bump, and the middle number bump for unstable releases.

But, all major version bumps are breaking changes in semver, which is more succinct and  aligned with the intent of the statement.

Reply:
> The reference to Semantic Versioning was to further explain what a "breaking change" was,

Yeah, that I understood. My point is that _disallowing_ such bumps (and/or breaking changes) should not be in the definition of "breaking change". Thus my suggestions
```suggestion
  - For projects following Semantic Versioning, major version bumps.
```
and alternatively
```suggestion
  - For projects following Semantic Versioning, major version bumps indicate breaking changes.
```

Reply:
Ah, I misunderstood

Reply:
applied

Reply:
Will commits of this PR be squashed before merging? If not you might want to fix the following:

61b40e8ea814f26d690ae348ab8a9a29b8d9c79d's commit message reads:
>  `Apply @davidak suggestion`

I guess autocomplete gave you the wrong username here. (I'm @das-g and the suggestion was by me.) For all I care, I don't need to be credited for that, though. Better describe what changed in the commit. (If suggestions for that are needed, I can come up with some.) If commits will be squashed and messages of intermediate commits discarded, don't bother, of course.